task-manager-backend/src/task/controllers/create-task/create-task.js

```diff
@@ -0,0 +1,46 @@
+const ApiResponse = require("../../../utils/api-response");
+const ApiError = require("../../../utils/api-error");
+const { createTaskUsecase } = require("../../usecases");
+const { sendMailUsecase } = require("../../../mail/usecases");
+
+const { TASK_CREATE_SUCCESS } = require("../../../commons/constants");
+
+async function sendTaskAssignEmail(userId, adminId) {
+    try {
+        await sendMailUsecase.execute("task_assign_email", userId, adminId);
+    } catch (error) {
+        throw error;
+    }
+}
+
+module.exports = {
+    createTask: {
+        path: "/",
+        reqType: "post",
+        method: async (req, res, next) => {
+            try {
+                const { title, dueDate, priority } = req.body;
+                const userId = req.session.isAdmin
+                    ? req.body.userId
+                    : req.session.userId;
+                const response = await createTaskUsecase.execute({
+                    title,
+                    dueDate,
+                    priority,
+                    userId,
+                });
+
+                if (
```

Comment:
No logic should be in this file. Since we are already passing the session can we implement this in sendTaskAssignmentEmail?

Reply:
I can remove equating condition, but if I remove the admin then the flow of sending mail iff admin is loggedin will get changed.

await sendTaskAssignEmail(userId, req.session);

calling function like this will make sense when mail needs to be sent but inside it will not send mail, if admin is not logged in.